homeassistant/components/generic/config_flow.py

```diff
@@ -170,26 +162,40 @@ async def async_test_still(
             response = await async_client.get(url, auth=auth, timeout=GET_IMAGE_TIMEOUT)
             response.raise_for_status()
             image = response.content
+            content_type = response.headers["content-type"]
     except (
         TimeoutError,
         RequestError,
         HTTPStatusError,
         TimeoutException,
+        KeyError,
     ) as err:
         _LOGGER.error("Error getting camera image from %s: %s", url, type(err).__name__)
         return {CONF_STILL_IMAGE_URL: "unable_still_load"}, None
-
     if not image:
         return {CONF_STILL_IMAGE_URL: "unable_still_load"}, None
-    fmt = get_image_type(image)
-    _LOGGER.debug(
-        "Still image at '%s' detected format: %s",
-        info[CONF_STILL_IMAGE_URL],
-        fmt,
-    )
-    if fmt not in SUPPORTED_IMAGE_TYPES:
+
+    # We can check everything apart from svg files.
+    if content_type != "image/svg+xml":
+        imagefile = io.BytesIO(image)
+        with contextlib.suppress(PIL.UnidentifiedImageError):
+            img = PIL.Image.open(imagefile)
+            fmt = "image/" + str(img.format.lower())
+            if fmt != content_type:
+                _LOGGER.warning(
+                    "Detected image content-type '%s' does not match the server reported content-type '%s'",
+                    fmt,
+                    content_type,
+                )
+            _LOGGER.debug(
+                "Still image at '%s' detected format: %s",
+                info[CONF_STILL_IMAGE_URL],
+                content_type,
+            )
+        content_type = fmt
```

Comment:
Isnt this same as the old function call. Seems logical to keep that then?

Reply:
It's doing much less now (no svg checking, as per your request).  There are only 3 lines left so I thought it was no longer worth a separate function.

Reply:
Ok. Its just the debug logging that makes it look larger than 3 lines